packages/medusa/src/services/shipping-option.ts

```diff
@@ -307,17 +412,19 @@ class ShippingOptionService extends BaseService {
       )
     }
 
-    const subtotal = cart.subtotal
-    const requirementResults = option.requirements.map((requirement) => {
-      switch (requirement.type) {
-        case "max_subtotal":
-          return requirement.amount > subtotal
-        case "min_subtotal":
-          return requirement.amount <= subtotal
-        default:
-          return true
+    const subtotal = cart.subtotal || 0
```

Comment:
I would maybe add a type guard here and throw a `UNEXPECTED_STATE` if the subtotal is not defined. 

The callee has the responsibility here to make sure that the cart has its subtotal calculated and I think it would be more appropriate to ensure that the responsibility is upheld by throwing an error.

Reply:
```suggestion
    if (typeof cart.subtotal === "undefined") {
      throw new MedusaError(MedusaError.Types.UNEXPECTED_STATE, "Cart subtotal must be defined for cart option validation")
    }
    const subtotal = cart.subtotal
```